### PR TITLE
add getRoleName function to get only single role name as string

### PR DESCRIPTION
**Pull request by SAM2904**
* Add new getRoleName function

### CODE REVIEW

**Comment by SAM2904** [comment]
This function will helps the developers to fetch the user role name without using  any array index key after function name.
It also helps to not trigger the error when rolename  is not exist at the specified index key (like as getRoleNames()[$key] function)

**Comment by parallels999** [comment]
>Add new getRoleName function 

You have to add tests for `getRoleName` function,
also function name must be more specific, using key means by key or by index?

> Improve code readability

Looks worst to me

**Comment by fredsal** [comment]
horrible

File: src/Traits/HasRoles.php [hunk 1]
```diff
@@ -314,6 +314,20 @@ public function getRoleNames(): Collection
...
+    /** Return only single rolename, developer don't need to use an arr key always.
+     * @param integer $key always sets to 0.
+     * @return String $role_name
+     */
+    public function getRoleName($key = 0): String
+    {
+        if (FALSE === is_int($key)) {
+            trigger_error('function getRoleName expected Argument 1 to be Integer', E_USER_WARNING);
+        }
+        if (is_int($key)) {
```
Comment (fredsal) [inline src/Traits/HasRoles.php:326]: Why?
unnecessary

File: src/Traits/HasRoles.php [hunk 2]
```diff
@@ -314,6 +314,20 @@ public function getRoleNames(): Collection
         return $this->roles->pluck('name');
     }
 
+    /** Return only single rolename, developer don't need to use an arr key always.
+     * @param integer $key always sets to 0.
+     * @return String $role_name
+     */
+    public function getRoleName($key = 0): String
+    {
+        if (FALSE === is_int($key)) {
```
Comment (fredsal) [inline src/Traits/HasRoles.php:323]: Always return a boolean, why `===`?

File: src/Traits/HasRoles.php [hunk 3]
```diff
@@ -314,6 +314,20 @@ public function getRoleNames(): Collection
...
+            trigger_error('function getRoleName expected Argument 1 to be Integer', E_USER_WARNING);
+        }
+        if (is_int($key)) {
+            return isset($this->roles[$key]->name) ? $this->roles[$key]->name : '';
```
Comment (fredsal) [inline src/Traits/HasRoles.php:327]: > isset($this->roles[$key]->name)
Don't seems to be right

Comment (SAM2904) [reply src/Traits/HasRoles.php]: can you please review it again @fredsal 
also, can you please guide me @fredsal to improve/fix this function.

**Comment by drbyte** [comment]
I'm not sure this needs to be in the package. It seems better to have this in userland application if one really needs it.

**Pull request closed without merging**